query_optimizer/ExecutionGenerator.cpp

```diff
@@ -477,14 +477,14 @@ void ExecutionGenerator::convertSelection(
   // Use the "simple" form of the selection operator (a pure projection that
   // doesn't require any expression evaluation or intermediate copies) if
   // possible.
-  std::unique_ptr<std::vector<attribute_id>> attributes(new std::vector<attribute_id>());
+  std::vector<attribute_id> attributes;
   SelectOperator *op
-      = convertSimpleProjection(project_expressions_group_index, attributes.get())
+      = convertSimpleProjection(project_expressions_group_index, &attributes)
         ? new SelectOperator(*input_relation_info->relation,
                              *output_relation,
                              insert_destination_index,
                              execution_predicate_index,
-                             attributes.release(),
+                             move(attributes),
```

Comment:
If you are talking about the shared reference between a operator and a work order, I think no. In fact, `SelectOperator` is the only one I need to change to run queries in the distributed version.